src/pages/Character/Character.js

```diff
@@ -0,0 +1,104 @@
+import React, { Component } from "react";
+import axios from "axios";
+import Layout from "../../components/Layout";
+import EpisodeCard from "../../components/EpisodeCard";
+import { getCharacter, getEpisodes } from "../../api";
+
+class Character extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      id: 0,
+      name: "",
+      image: "",
+      species: "",
+      status: "",
+      origin: "",
+      location: "",
+      episodes: [],
+      hasLoaded: false,
+      hasError: null,
+    };
+    this.loadCharacter = this.loadCharacter.bind(this);
+  }
+
+  componentDidMount() {
+    const { match } = this.props;
+    const { characterId } = match.params;
+
+    this.loadCharacter(characterId);
+  }
+
+  async loadCharacter(characterId) {
```

Comment:
Yo me imagino, Ethan, que de algun modo te ha salido bien todo la pill. Solo un consejito. Dentro de esta función, siento que te falta un TRY CACTH, para que te devuelva un error en caso de no poder hacer la petición fetch. Ya que veo que has hecho todo el sistema sacando las variables  con el axios.get() y recorriendolo con un .map()

Ejemplo de como lo hice (no te fijes tanto en como hice la peticion de Apis, mas bien solo en la estructura del try cacth.)
    try {
      const { data } = await getCharacters(characterId);

      // eslint-disable-next-line compat/compat
      const characterEpisodes = await Promise.all(
        makePromises(data.episode),
      );

      const episodesFromCharacter = characterEpisodes.map((character) => character.data);

      console.log(episodesFromCharacter);

      this.setState({
        episodes: episodesFromCharacter,
        name: data.name,
        image: data.image,
        species: data.species,
        status: data.status,
        origin: data.origin.name,
        location: data.location.name,
        hasLoaded: true,
        hasError: false,
        errorMessage: null,

      });

      console.log(data);
    } catch (error) {
      this.setState({
        hasLoaded: true,
        hasError: true,
        errorMessage: error.message,
      });
    }

De esta manera, al final, en el catch () es donde se puede setear el estado de mejor manera.